File: 步进电机SMD258C_code/RaspberryPi/python/MotorDriver.py
import RPi.GPIO as GPIO
import time

MotorDir = [
    'forward',
    'backward',
]

class MotorDriver():
    def __init__(self, mode):
        self.dir_pin = 13
        self.step_pin = 19        
        self.enable_pin = 12
        self.mode = mode
        # mode = 0 : Positive PU- DR- MF- to GND
        # mode = 1 : Negative PU- DR- MF- to 5V
        
        GPIO.setmode(GPIO.BCM)
        GPIO.setwarnings(False)
        GPIO.setup(self.dir_pin, GPIO.OUT)
        GPIO.setup(self.step_pin, GPIO.OUT)
        GPIO.setup(self.enable_pin, GPIO.OUT)
                
    def digital_write(self, pin, value):
        GPIO.output(pin, value)
        
    def Stop(self):
        self.digital_write(self.enable_pin, ~self.mode)
    
    def TurnStep(self, Dir, steps, stepdelay=0.002):       
        if (steps == 0):
            return
            
        self.digital_write(self.enable_pin, self.mode)
        if self.mode == 0 :
            self.digital_write(self.dir_pin, 0 if (Dir == MotorDir[0]) else 1)
        else:
            self.digital_write(self.dir_pin, 1 if (Dir == MotorDir[0]) else 0)
            
        print "turn step:",steps
        for i in range(steps):
            self.digital_write(self.step_pin, False)
            time.sleep(stepdelay)
            self.digital_write(self.step_pin, True)
            time.sleep(stepdelay)

File: 步进电机SMD258C_code/Arduino/UNO R3/demo/MotorDriver.cpp
/*****************************************************************************
  | File        :   MotorDriver.cpp
  | Author      :   Waveshare team
  | Function    :
  | Info        :

  ----------------
  |	This version:   V1.0
  | Date        :   2018-10-13
  | Info        :   Basic version

******************************************************************************/
#include "MotorDriver.h"
#include "Debug.h"  //DEBUG()
MOTOR Motor;

/**
   Select motor

   @param name: motor.

   Example:
   Motor_Init();

*/
void Motor_Init(void)
{
  Motor.EnablePin = ENABLE_PIN;
  Motor.DirPin = DIR_PIN;
  Motor.StepPin = STEP_PIN;

  // PU- DR- MF- to GND
  Motor.Positive = 1;
  Motor.Negative = 0;

  // PU+ DR+ MF+ to VCC
  // Motor.Negative = 0;
  // Motor.Positive = 1;
}

/**
   The motor stops rotating and the driver chip is disabled.

*/
void Motor_Stop(void)
{
  DEV_Digital_Write(Motor.EnablePin, Motor.Positive);
}

/**
   turn.

   @param dir: direction.
   @param steps: Step count.
   @param stepdelay: step delay.

   Example:
   Motor_TurnStep()
*/
void Motor_TurnStep(UBYTE dir, UWORD steps, UWORD stepdelay)
{
  if (dir == FORWARD) {
    DEV_Digital_Write(Motor.EnablePin, Motor.Negative);
    DEV_Digital_Write(Motor.DirPin, Motor.Negative);
  } else if (dir == BACKWARD) {
    DEV_Digital_Write(Motor.EnablePin, Motor.Negative);
    DEV_Digital_Write(Motor.DirPin, Motor.Positive);
  } else {
    DEV_Digital_Write(Motor.EnablePin, Motor.Positive);
  }

  if (steps == 0)
    return;

  UWORD i = 0;
  for (i = 0; i < steps; i++) {
    DEV_Digital_Write(Motor.StepPin, 1);
    DEV_Delay_ms(stepdelay);
    DEV_Digital_Write(Motor.StepPin, 0);
    DEV_Delay_ms(stepdelay);
  }

}
